rclcpp/test/test_time.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "rcl/error_handling.h"
+#include "rcl/time.h"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/time.hpp"
+
+class TestTime : public ::testing::Test
+{
+protected:
+  static void SetUpTestCase()
+  {
+    rclcpp::init(0, nullptr);
+  }
+};
+
+TEST(TestTime, rate_basics) {
+  using builtin_interfaces::msg::Time;
+  try {
+    rclcpp::Time::now<RCL_ROS_TIME>();
```

Comment:
Should there be `RCLCPP_*_TIME` classes to avoid this mixing of rclcpp and rcl things?